cmake/Git.cmake

```diff
@@ -9,6 +9,10 @@ if (GIT_INFO)
 
     option(INCLUDE_GIT_COMMIT_HASH "Embed the latest git commit abbreviated hash into the sources." ON)
 
+    if(INCLUDE_GIT_COMMIT_HASH)
+        add_definitions(-DINCLUDE_GIT_COMMIT_HASH)
```

Comment:
toto nie je idealne, pretoze add_definitions() to nastavi zbytocne pre vsetky *.cpp 
a tento define je potrebny len v main.cpp (je to zastaraly pristup)
- vid. pouzitie premennej `ALL_COMPILE_DEFINITIONS` a target_compile_definitions()

lepsie riesenie je upravit
`target_compile_definitions( ${3DSOFTVIZ_NAME}_app PUBLIC ${ALL_COMPILE_DEFINITIONS} )`
na 
`target_compile_definitions( ${3DSOFTVIZ_NAME}_app PUBLIC ${ALL_COMPILE_DEFINITIONS} -DINCLUDE_GIT_COMMIT_HASH )`
resp. (ale to nastavi tento define aj pre 3dsoftviz.lib, co netreba)
```
if(INCLUDE_GIT_COMMIT_HASH)
   list(APPEND ALL_COMPILE_DEFINITIONS -DINCLUDE_GIT_COMMIT_HASH)
endif ()
```

poznamka:
- ALL_COMPILE_DEFINITIONS nie je dobre riesenie - vramci refaktoringu by sa malo upravit

Reply:
treba s tymto pohnut

Reply:
ked uz nechceme mat definicie viditelne vsade navrhujem skor 
        set_property(SOURCE ${CMAKE_SOURCE_DIR}/src/main.cpp APPEND PROPERTY COMPILE_DEFINITIONS INCLUDE_GIT_COMMIT_HASH)
co by teoreticky malo nastavit define iba pre ten subor